tests/virt/upgrade/utils.py

```
@@ -29,11 +34,10 @@
 TIMESTAMP_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
 
 
-def get_virt_launcher_image_from_csv(csv):
-    for item in csv.instance.spec.relatedImages:
-        if VIRT_LAUNCHER in item["name"]:
-            return item["image"]
-    raise ValueError(f"Image digest for {VIRT_LAUNCHER} not found")
+def get_virt_launcher_images_from_csv(csv: ClusterServiceVersion) -> set[str]:
+    if images := {item["image"] for item in csv.instance.spec.relatedImages if VIRT_LAUNCHER in item["name"]}:
+        return images
+    raise ValueError(f"Image digests for {VIRT_LAUNCHER} not found")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)